src/libstd/io/buffered.rs

```diff
@@ -24,7 +24,8 @@ use crate::memchr;
 ///
 /// When the `BufReader<R>` is dropped, the contents of its buffer will be
 /// discarded. Creating multiple instances of a `BufReader<R>` on the same
-/// stream can cause data loss.
+/// stream can cause data loss, as well as reading from the underlying reader
```

Comment:
How about:

_Reading from the underlying reader after unwrapping the `BufReader<R>` with `BufReader::into_inner` can also cause data loss._